packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -41,6 +43,13 @@ export interface BucketDeploymentProps {
    */
   readonly retainOnDelete?: boolean;
 
+  /**
+   * A map of object metadata to set on the specified objects
+   *
+   * @default - No object metadata is set
+   */
+  readonly objectsMetadata?: ObjectsMetadata;
```

Comment:
Is there really a use case for setting different metadata on different keys? Isn't it possible to just define multiple `BucketDeployment`s for each type of object? Feels like this convolutes the API dramatically.

Furthermore, I am wondering if it might be possible to offer a richer API for the system-level metadata (for example, I'd expect `expires` to be of type `Duration` and not just a string).

Reply:
Agreed. I've simplified the API now and offered a richer API for the system-defined metadata.

Reply:
Split system and user to to top-level properties. We prefer less nesting. Maybe even the system one could be just flat fields at the top level struct without a group (kind of like how the AWS CLI has them as first class switches)